app/src/androidTest/java/com/github/campus_capture/bootcamp/ZoneDAOTest.java

```diff
@@ -53,4 +54,95 @@ public void writeZoneAndReadInList() throws Exception {
     }
+
+    @Test
+    public void writeZoneAndGetAll() throws Exception {
+        List<LatLng> vertices1 = new ArrayList<>();
+        vertices1.add(new LatLng(10, 20));
+        vertices1.add(new LatLng(0, 0));
+        vertices1.add(new LatLng(-1000, 20.1));
+
+        List<LatLng> vertices2 = new ArrayList<>();
+        vertices2.add(new LatLng(10, 2));
+        vertices2.add(new LatLng(9, 0));
+        vertices2.add(new LatLng(-100, -10));
+
+        Zone zone1 = new Zone("zone1", vertices1);
+        Zone zone2 = new Zone("zone2", vertices2);
+
+        List<Zone> zoneArray = new ArrayList();
+        zoneArray.add(zone1);
+        zoneArray.add(zone2);
+
+        zoneDAO.insertAll(zone1, zone2);
+
+        List<Zone> zoneArrayDB = zoneDAO.getAll();
+
+        for(int i = 0; i < zoneArray.size(); ++i){
+            assertThat(zoneArray.get(i).equals(zoneArrayDB.get(i)), equalTo(true));
+        }
+    }
+
+    @Test
+    public void writeZoneAndloadAll() throws Exception {
+        List<LatLng> vertices1 = new ArrayList<>();
+        vertices1.add(new LatLng(10, 20));
+        vertices1.add(new LatLng(0, 0));
+        vertices1.add(new LatLng(-1000, 20.1));
+
+        List<LatLng> vertices2 = new ArrayList<>();
+        vertices2.add(new LatLng(10, 2));
+        vertices2.add(new LatLng(9, 0));
+        vertices2.add(new LatLng(-100, -10));
+
+        Zone zone1 = new Zone("zone1", vertices1);
+        Zone zone2 = new Zone("zone2", vertices2);
+
+        List<Zone> zoneArray = new ArrayList();
+        zoneArray.add(zone1);
+        zoneArray.add(zone2);
+
+        zoneDAO.insertAll(zone1, zone2);
+
+        int [] zoneList = IntStream.rangeClosed(0, zoneArray.size()).toArray();
+
+        List<Zone> zoneArrayDB = zoneDAO.loadAllByIds(zoneList);
+
+        for(int i = 0; i < zoneArray.size(); ++i){
+            assertThat(zoneArray.get(i).equals(zoneArrayDB.get(i)), equalTo(true));
+        }
+    }
+
+    @Test
+    public void writeZoneAndDelete() throws Exception {
+        List<LatLng> vertices1 = new ArrayList<>();
+        vertices1.add(new LatLng(10, 20));
+        vertices1.add(new LatLng(0, 0));
+        vertices1.add(new LatLng(-1000, 20.1));
+
+        List<LatLng> vertices2 = new ArrayList<>();
+        vertices2.add(new LatLng(10, 2));
+        vertices2.add(new LatLng(9, 0));
+        vertices2.add(new LatLng(-100, -10));
+
+        Zone zone1 = new Zone("zone1", vertices1);
+        Zone zone2 = new Zone("zone2", vertices2);
+
+        List<Zone> zoneArray = new ArrayList();
+        zoneArray.add(zone1);
+        zoneArray.add(zone2);
+
+        zoneDAO.insertAll(zone1, zone2);
+
+        List<Zone> zoneArrayDB = zoneDAO.getAll();
+
+        for(int i = 0; i < zoneArray.size(); ++i){
+            assertThat(zoneArray.get(i).equals(zoneArrayDB.get(i)), equalTo(true));
+        }
+
+        zoneDAO.delete(zone1);
+
+        assertThat(zoneDAO.findByName("zone2").equals(zone2), equalTo(true));
```

Comment:
I understand that your test is checking that both zones are present before the deletion, and then checking that the second zone is present, but you should also check if the zone1 is absent. I don't know if it is meant to throw an exception or return null if you search a non-existent zone by name, but it makes sense to add a check to see if the first zone really has been deleted.

Reply:
Your're right, I'm adding it

Reply:
Update : You're right, the zone is not actually deleted (!)
Thank you for pointing this out. I'm working on a fix now